Fix bytes.isspace

hbina opened the pull request:
Fix implementation of bytes.isspace to match the implementation in CPython.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved `bytes.isspace()` handling for vertical tabs and other ASCII whitespace.
  * Corrected `bytes.istitle()` behavior for capitalization, separators, digits, whitespace, and empty values.

* **Tests**
  * Added comprehensive coverage for whitespace and title-case byte strings across common encodings.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

hbina committed:
Fix implementation of bytes.isspace to match the implementation in CPython.

youknowone commented on vm/src/bytesinner.rs:
```diff
@@ -363,11 +363,18 @@ impl PyBytesInner {
     }
 
     pub fn isspace(&self) -> bool {
+        // What CPython considers whitespace is a bit different from what Rust.
```
👍

hbina committed:
Fix implementation of bytes.istitle so that test_bigmem now passes.
Also added some extra_test to capture this.

hbina commented on Lib/test/test_bigmem.py:
```diff
@@ -786,17 +786,6 @@ def test_title(self, size):
-    # TODO: RUSTPYTHON
-    @unittest.expectedFailure
-    @bigmemtest(size=_2G, memuse=2)
-    def test_istitle(self, size):
```
Ok I have pushed a commit that fixes istitle.

hbina committed:
Fix some clippy issues

Signed-off-by: Hanif Ariffin <hanif.ariffin.4326@gmail.com>

hbina committed:
bytes.istitle returns false when the bytes is empty.

youknowone commented:
`test_bytes` is not passing
```
======================================================================
FAIL: test_istitle (test.test_bytes.ByteArrayAsStringTest.test_istitle)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/string_tests.py", line 968, in test_istitle
    self.checkequal(False, 'Not--a Titlecase String', 'istitle')
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/string_tests.py", line 68, in checkequal
    realresult
AssertionError: False != True

======================================================================
FAIL: test_istitle (test.test_bytes.BytesAsStringTest.test_istitle)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/string_tests.py", line 968, in test_istitle
    self.checkequal(False, 'Not--a Titlecase String', 'istitle')
  File "/home/runner/work/RustPython/RustPython/pylib/Lib/test/string_tests.py", line 68, in checkequal
    realresult
AssertionError: False != True
```

hbina committed:
Fix some edges of istitle with non-alphanumeric characters

hbina committed:
Disable a line because python 3.12 errors on bad escape sequence

hbina committed:
Match CPython's istitle a little closer

hbina committed:
Fix clippy warning

youknowone committed:
Rewrite bytes.istitle along _Py_bytes_istitle

The lookahead-based check accepted b"Not--a Titlecase String", which
test_bytes covers. Track whether the previous byte was cased instead.

isspace compares bytes directly rather than converting to char.

Restore the b"omkmok\Xaa" assert: 3.14 reports an invalid escape
sequence as a SyntaxWarning, not an error.

Assisted-by: Claude
